### ISSUE TITLE
dataset problem

### ISSUE THREAD

**Issue by sola20** [QinLab-WFU/DSPH#2]
Hello, after reviewing the code, it seems that the data input needs to input images based on the index path (e.g. root_dir \ actor \ image. jpg). However, the file source does not seem to contain the actual image, but rather a compilation of the image list and its corresponding URL. Can the author provide a compressed file containing the original Nuswide image set used by the author in this article? Sincere thanks!

**Comment by 1zx3** [comment]
hello，Same question here， did you solve it？

**Comment by sola20** [comment]
Yes, after sending an email to inquire, the author provided the link to download the original images from nuswide: https://www.kaggle.com/datasets/twerwweqweq/nuswide  or  https://www.kaggle.com/datasets/ssaysham/nuswide-for-deephash

**Comment by 1zx3** [comment]
Hello, thank you very much for providing the link, but this situation still exists after I imported the images. Do I need to classify multi label images? Or other reasons? If possible, could you provide the classified dataset? Sincere thanks
The link to the original text is images/images. jpg. There is no ship file like the one below
![1](https://github.com/QinLab-WFU/DSPH/assets/169539078/9fc50406-5949-4ce0-aecd-8cf74b08df8c)

**Comment by sola20** [comment]
I don't think it's necessary to classify multi label images. In fact, the image import steps are not related to the labels. My approach is to batch modify the image path of each line in the imagelist.txt file in Python, remove the category, and modify the path to the correct path to ensure that the image can be found after concatenating the index path generated in the make_cuswide.py file.

**Comment by 1zx3** [comment]
I'll give it a try, thank you very much again

**Comment by 1zx3** [comment]
I'll give it a try, thank you very much again


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "QinLab-WFU/DSPH"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年7月5日(星期五) 晚上10:20
***@***.***&gt;;
抄送:&nbsp;"Gao ***@***.******@***.***&gt;;
主题:&nbsp;Re: [QinLab-WFU/DSPH] dataset problem (Issue #2)





 
I don't think it's necessary to classify multi label images. In fact, the image import steps are not related to the labels. My approach is to batch modify the image path of each line in the imagelist.txt file in Python, remove the category, and modify the path to the correct path to ensure that the image can be found after concatenating the index path generated in the make_cuswide.py file.

**Comment by QinLab-WFU** [comment]
We recognize the previous link fails, nuswide images: https://pan.baidu.com/s/11IxoukhNPvypItt014k6qQ 提取码：3g1g

**Comment by Ypetrol** [comment]
Hello, when conducting experiments with the NUS-WIDE dataset, the MAP value remains NaN and 0 after calculation. How can I solve this issue?

**Comment by 1zx3** [comment]
> 您好，在使用 NUS-WIDE 数据集进行实验时，MAP 值计算后始终为 NaN 和 0。请问该如何解决？
The reason may be that the learning rate (lr) is too high or too low

**Comment by Ypetrol** [comment]
Could you tell me the learning rate used for training the NUS-WIDE dataset in the paper “Deep multi-similarity hashing via label-guided network for cross-modal retrieval”? Thank you very much.

**Comment by 1zx3** [comment]
I have only encountered similar problems before and do not know the detailed settings of the literature you mentioned.

**Comment by Rooike111** [comment]
> Hello, when conducting experiments with the NUS-WIDE dataset, the MAP value remains NaN and 0 after calculation. How can I solve this issue?

Some images in Nuswide data have no classification at all, so this situation sometimes occurs. You need to handle it with matlap.
